src/lint_problems.ts

```diff
@@ -0,0 +1,279 @@
+        const normalizedPath = normalizePath(absolutePath);
+        if (!isPathNeuroSafe(normalizedPath)) {
+            return actionResultNoAccess(normalizedPath);
+        }
+        if (!fs.existsSync(normalizedPath)) {
+            return actionResultFailure(`File '${relativePath}' does not exist.`);
+        }
+
+        const rawDiagnostics = vscode.languages.getDiagnostics(vscode.Uri.file(normalizedPath));
+        const formattedDiagnostics = getFormattedDiagnosticsForFile(relativePath, rawDiagnostics); // use relativePath
+        NEURO.client?.sendContext(`Linting problems for file ${relativePath}:${formattedDiagnostics}`);
+    } catch (error) {
+        NEURO.client?.sendContext(`Failed to get linting diagnostics for '${relativePath}'.`);
+        logOutput("ERROR", `Getting diagnostics for ${relativePath} failed: ${error}`);
+    }
+    return actionResultAccept();
+}
+
+
+// Handle diagnostics for a folder
+export function handleGetFolderLintProblems(actionData: ActionData): ActionResult {
+    if (!hasPermissions(PERMISSIONS.accessLintingAnalysis)) {
+        return actionResultNoPermission(PERMISSIONS.accessLintingAnalysis);
+    }
+    const relativeFolder = actionData?.params.folder;
+    if (!relativeFolder) {
+        return actionResultMissingParameter('folder');
+    }
+
+    const workspacePath = getWorkspacePath();
+    if (!workspacePath) {
+        logOutput("ERROR", 'Workspace folder not found.');
+        return actionResultFailure("Unable to get workspace path.");
+    }
+
+    try {
+        const absoluteFolderPath = path.join(workspacePath, relativeFolder);
+        const normalizedFolderPath = normalizePath(absoluteFolderPath);
+        if (!isPathNeuroSafe(normalizedFolderPath)) {
+            return actionResultNoAccess(normalizedFolderPath);
+        }
+        if (!fs.existsSync(normalizedFolderPath)) {
+            return actionResultFailure(`Folder '${relativeFolder}' does not exist.`);
+        }
+
+        const diagnostics = vscode.languages.getDiagnostics();
+        const folderDiagnostics = diagnostics.filter(([uri]) => {
+            return uri.fsPath.startsWith(normalizedFolderPath) && isPathNeuroSafe(uri.fsPath);
+        });
+
+        if (folderDiagnostics.length === 0) {
+            return actionResultNoAccess(normalizedFolderPath);
+        }
+
+        const formattedDiagnostics = folderDiagnostics.map(([uri, diags]) => {
+            const relative = path.relative(workspacePath, uri.fsPath);
+            return getFormattedDiagnosticsForFile(relative, diags);
+        }).join('\n');
+
+        NEURO.client?.sendContext(`Linting problems for folder ${relativeFolder}: ${formattedDiagnostics}`);
+    } catch (error) {
+        NEURO.client?.sendContext(`Failed to get linting diagnostics for folder '${relativeFolder}'.`);
+        logOutput("ERROR", `Getting diagnostics for folder ${relativeFolder} failed: ${error}`);
+    }
+    return actionResultAccept();
```

Comment:
There is no need for separate context as everything is synchronous
```suggestion
        return actionResultAccept(`Linting problems for folder ${relativeFolder}: ${formattedDiagnostics}`);
    } catch (error) {
        logOutput("ERROR", `Getting diagnostics for folder ${relativeFolder} failed: ${error}`);
        return actionResultFailure(`Failed to get linting diagnostics for folder '${relativeFolder}'.`);
    }
```